README.md

```diff
@@ -112,84 +125,141 @@ The content of your document is retrieved.
+
 #### keyLength
 
+The key size should be at least 128 bits. See this [NIST publication](http://csrc.nist.gov/publications/nistpubs/800-38D/SP-800-38D.pdf) to learn more.
+
 #### ivLength
 
+In Galois/Counter Mode (GCM), a 12 byte IV is strongly suggested as other IV lengths will require additional calculations. 
+
+In principle, any IV size can be used as long as the IV doesn't repeat. However, [NIST](http://csrc.nist.gov/publications/nistpubs/800-38D/SP-800-38D.pdf) suggests only an IV size of 12 bytes needs to be supported in the implementations.
+
 #### tagLength
 
+Use a tag size of 64 bits at the very minimum, but it is preferable to use a tag size of the full 128 bits.
+
 #### version
 
+<!--Flag: Through a search in the NIST publication and the post on stackexchange, version was not found. We need to tell the users which version it is talking about.-->
+
 ### generateEncryptionKey
 
-Generates a random key represented as a hexadecimal string
+This request generates a random key represented as a hexadecimal string.
 
 #### Parameters
 
-*   `keyLengthInBits` **[number](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Number)** Key length (optional, default `ENCRYPTION_PARAMETERS.keyLength`)
+This request supports the parameter below:
+
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|-------------|
+|`keyLengthInBits`|**[Number](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Number)**| Optional  | Default `ENCRYPTION_PARAMETERS.keyLength` |
+
 
 ### encodeDocument
 
+This request performs encoding on a document.
+
 #### Parameters
 
-*   `document` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**&#x20;
+This request supports the parameter below:
+<!--Flag: pls check the necessity column, if it is optional or required.-->
+
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|-------------|
+|`document`|**[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**| Optional | Provide a document to encode it |
 
 ### decodeDocument
 
+This request performs decoding on an encoded document.
+
 #### Parameters
 
-*   `encoded` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**&#x20;
+This request supports the parameter below:
+<!--Flag: pls check the necessity column, if it is optional or required.-->
+
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|-------------|
+|`encoded`|**[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**| Optional | Provide an encoded document to decode it |
 
 ### IEncryptionResults
 
-### encryptString
+<!--Flag: pls clarify if we can add some description to explain IEncryptionResults -->
```

Comment:
you can refer to my earlier comment,`IEncryptionResults` is the interface/type returned by `encryptString` and expected by `decryptString`

Reply:
Noted